cmd/zoekt-sourcegraph-indexserver/index.go

```diff
@@ -209,7 +209,15 @@ func gitIndex(o *indexArgs, runCmd func(*exec.Cmd) error) error {
 	// clone. So don't defer os.RemoveAll here
 
 	// Create a repo to fetch into
-	cmd := exec.CommandContext(ctx, "git", "init", "--bare", gitDir)
+	cmd := exec.CommandContext(ctx, "git",
+		// use a random default branch. This is so that HEAD isn't a symref to a
+		// branch that is indexed. For example if you are indexing
+		// HEAD,master. Then HEAD would be pointing to master by default.
+		"-c", "init.defaultBranch=BB0FOFCH32",
```

Comment:
maybe `sourcegraphNonexistentBranch` to be more obvious to a future debugger?

Reply:
I thought about using something like that, but what if that was chosen! I'll use that as a prefix :)